src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/TextFieldDisplayDriver.cs

```diff
@@ -38,7 +38,8 @@ public override IDisplayResult Edit(TextField field, BuildFieldEditorContext con
         {
             return Initialize<EditTextFieldViewModel>(GetEditorShapeType(context), model =>
             {
-                model.Text = field.Text;
+                var settings = context.PartFieldDefinition.GetSettings<TextFieldSettings>();
+                model.Text = context.IsNew ? settings.DefaultValue : field.Text;
```

Comment:
If we were to take this PR,  you may want to change this line to 

model.Text = context.IsNew && field.Text == null ? settings.DefaultValue : field.Text;

Reply:
Implemented with newest commit